packages/roosterjs-editor-dom/lib/userPosition/adjustUserPosition.ts

```diff
@@ -0,0 +1,238 @@
+import {
+    EditorCore,
+    NodePosition,
+    NodeType,
+    PositionType,
+    QueryScope,
+} from 'roosterjs-editor-types';
+import {
+    changeElementTag,
+    contains,
+    createRange,
+    findClosestElementAncestor,
+    getBlockElementAtNode,
+    getTagOfNode,
+    isNodeEmpty,
+    isPositionAtBeginningOf,
+    isVoidHtmlElement,
+    LinkInlineElement,
+    Position,
+    PositionContentSearcher,
+    queryElements,
+    splitBalancedNodeRange,
+    splitTextNode,
+    toArray,
+    unwrap,
+    VTable,
+    wrap,
+} from 'roosterjs-editor-dom';
+
+/**
+ * Adjust position for <a> tag don't be nested inside another <a> tag.
```

Comment:
For some reason, let's encode all "<" and ">" to be &lt; and &gt; in comment so that it can generate better document by typedoc.